src/AggregateFunctions/AggregateFunctionNull.h

```diff
@@ -414,6 +414,109 @@ class AggregateFunctionNullVariadic final
         this->nested_function->add(this->nestedPlace(place), nested_columns, row_num, arena);
     }
 
+    void addBatchSinglePlace(
+        size_t row_begin,
+        size_t row_end,
+        AggregateDataPtr __restrict place,
+        const IColumn ** columns,
+        Arena * arena,
+        ssize_t if_argument_pos) const final
+    {
+        /// We are going to merge all the flags into a single one to be able to call the nested batching functions
+        std::vector<const UInt8 *> nullable_filters;
+        const IColumn * nested_columns[number_of_arguments];
+
+        std::unique_ptr<UInt8[]> final_flags = nullptr;
+        const UInt8 * final_flags_ptr = nullptr;
+
+        if (if_argument_pos >= 0)
+        {
+            final_flags = std::make_unique<UInt8[]>(row_end);
+            final_flags_ptr = final_flags.get();
+
+            bool included_elements = 0;
+            const auto & flags = assert_cast<const ColumnUInt8 &>(*columns[if_argument_pos]).getData();
+            for (size_t i = row_begin; i < row_end; i++)
+            {
+                final_flags[i] = !flags.data()[i];
```

Comment:
the index for accessing to array `final_flags` is between row_begin and row_end , I think the [0, row_begin) part of the array will never be accessed.